### DISCUSSION TITLE
how to use start my express aplication

### DISCUSSION THREAD

**Discussion by Bobonimo111** [community/community#144723] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

hey, i am new in this community and a i have a question, how i config git hub pages to start my express application in ts and run a exemplo in pages.

**Comment by BancoCentraI** [comment, marked as answer]
Configurar o Backend (Express com TypeScript)
Crie o projeto Express com TypeScript:

mkdir express-ts-app
cd express-ts-app
npm init -y
npm install express
npm install typescript ts-node @types/node @types/express --save-dev
Crie o arquivo tsconfig.json:


{
  "compilerOptions": {
    "target": "ES6",
    "module": "CommonJS",
    "outDir": "./dist",
    "rootDir": "./src",
    "strict": true
  },
  "include": ["src/**/*"]
}
Crie o arquivo src/index.ts:


import express from 'express';
const app = express();
const port = 3000;

app.get('/', (req, res) => {
  res.send('Hello from Express!');
});

app.listen(port, () => {
  console.log(`Server running on http://localhost:${port}`);
});
Execute o servidor:


npx ts-node src/index.ts
2. Configurar o Frontend com GitHub Pages
Crie a pasta public e adicione o arquivo index.html com o conteúdo do seu frontend.

Instale o gh-pages para deploy:

npm install gh-pages --save-dev
Adicione scripts no package.json:


"scripts": {
  "start": "node dist/index.js",
  "build": "tsc",
  "deploy": "npm run build && gh-pages -d dist"
}
Faça o deploy para o GitHub Pages:


npm run deploy
3. Hospedar o Backend (Express)
Escolha uma plataforma para hospedar o backend (como [Heroku](https://www.heroku.com/), [Vercel](https://vercel.com/), ou [Railway](https://railway.app/)).
Siga os tutoriais dessas plataformas para configurar e fazer o deploy do seu backend Express.
Resumo:
GitHub Pages vai hospedar seu frontend (arquivos estáticos).
Seu backend Express (em TypeScript) precisa ser hospedado em outro serviço como Heroku ou Vercel.
